packages/transaction-controller/src/types.ts

```diff
@@ -1480,6 +1485,9 @@ export type SimulationData = {
 
   /** Data concerning a change to the user's token balances. */
   tokenBalanceChanges: SimulationTokenBalanceChange[];
+
+  /** The gas used by the transaction in the simulation. */
+  gasUsed?: Hex;
```

Comment:
Since we're storing this in a dedicated property, can we avoid conflating it with the simulation data which is specific to balance changes and so we don't persist it twice?